### PR TITLE
Feature add integration tests for easy testing and verification of APIs.

### PR DESCRIPTION
**Pull request by chris-cmsoft**
* Adds integration test suites to test full stateful endpoints of the service. This significantly eases verification of APIS and development processes
* Adds a GetPlan endpoint which was needed on the UI

### CODE REVIEW

File: api/handler/plan_api_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		planHandler := NewPlanHandler(logger.Sugar(), service.NewPlanService(bus.Publish))
+		e := echo.New()
+
+		// Setup
+		reqBody, _ := json.Marshal(map[string]interface{}{
+			"title": "Some Plan",
+		})
+		req := httptest.NewRequest(
+			http.MethodPost,
+			"/plan",
```
Comment (harrylincoln-CS) [inline api/handler/plan_api_test.go:69]: try changing this to "/not-a-plan" - the test still passes.

It looks like it's creating a simulated HTTP request object, and the test logic directly invokes the CreatePlan or GetPlan handler functions. Could it be that echo isn't configured to "listen" for these calls?

Comment (harrylincoln-CS) [reply api/handler/plan_api_test.go]: @chris-cmsoft I've added a commit to demonstrate what I mean by how requests must pass through Echo's router 👌

File: api/handler/plan_api_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	tests.IntegrationTestSuite
+}
+
+func (suite *PlanIntegrationSuite) TestCreatePlan() {
+	suite.Run("A plan can be created through the API", func() {
+		e := echo.New()
+		logger, _ := zap.NewProduction()
+		planHandler := NewPlanHandler(logger.Sugar(), service.NewPlanService(bus.Publish))
+
+		e.POST("/plan", planHandler.CreatePlan)
```
Comment (chris-cmsoft) [inline api/handler/plan_api_test.go:37]: This is a good idea for us to add the actual POST piece. Maybe we can just register the handlers routes using the server function, and creating a group called `api/plan` ?

Comment (chris-cmsoft) [reply api/handler/plan_api_test.go:37]: https://github.com/compliance-framework/configuration-service/pull/142/files#diff-63de9eb2ef299afee6ce75f0fc1155b74548ffda5759f7d57c3e0c7ef851fe97R20

**Review by harrylincoln-CS — APPROVED** [review]

**Pull request merged**